### ISSUE TITLE
Add JavaDoc to CaseFunction

### ISSUE THREAD

**Issue by donraab** [eclipse/eclipse-collections#1456]
CaseFunction currently has no JavaDoc. It should clarify that the cases are evaluated in order and the first match results in a return value. Multiple cases are not supported.

**Comment by ShashidharYes** [comment]
Hello, @donraab I'm a beginner in open source, I am willing to work on this issue can you please guide me on how I can get started?

**Comment by ShashidharM0118** [comment]
Hello, @donraab I'm a beginner in open source, I am willing to work on this issue, can you please guide me on how I can get started?

**Comment by relentless-pursuit** [comment]
> CaseFunction currently has no JavaDoc. It should clarify that the cases are evaluated in order and the first match results in a return value. Multiple cases are not supported.

True. Can I proceed with this? Or, Maybe have a draft PR which you can evaluate?

**Comment by sharath2106** [comment]
Hi @relentless-pursuit . Are you working on this? If not, i can pick this up. Please let me know.

**Comment by kush2439p** [comment]
I am working on this and will submit a focused PR that documents CaseFunction evaluation order, first-match behavior, and the single-result semantics described in the issue.

**Comment by kush2439p** [comment]
Withdrawing my claim after reviewing the repository's AI contribution policy; I will not submit a PR for this issue, so it remains available to other contributors.
